From 564356109fcedbb5e517276633ff1a1bb1da57a0 Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Thu, 9 Aug 2018 17:44:35 +0800
Subject: [PATCH 4367/5242] MLK-18772: ISI: correct isi RGB to YUV conversion
 coefficients

commit  04ad124f74619760f0d12557acacb24fbc65ff74 from
https://source.codeaurora.org/external/imx/linux-imx.git

Correct RGB to YUV conversion coefficients and change csc mode
to RGB to YCbCr

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Reviewed-by: Robby.Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-isi-hw.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-isi-hw.c b/drivers/media/platform/imx8/mxc-isi-hw.c
index 464a3af..a79c1a6 100644
--- a/drivers/media/platform/imx8/mxc-isi-hw.c
+++ b/drivers/media/platform/imx8/mxc-isi-hw.c
@@ -71,7 +71,7 @@ void dump_isi_regs(struct mxc_isi_dev *mxc_isi)
 	/* YUV2RGB */
 	{ 0x0000012A, 0x012A0198, 0x0730079C, 0x0204012A, 0x01F00000, 0x01800180 },
 	/* RGB->YUV */
-	{ 0x0096004D, 0x05DA001D, 0x007005B6, 0x057C009E, 0x000005E6, 0x00000000 },
+	{ 0x00810041, 0x07db0019, 0x007007b6, 0x07a20070, 0x001007ee, 0x00800080 },
 };
 
 static void printk_pixelformat(char *prefix, int val)
@@ -292,7 +292,7 @@ void mxc_isi_channel_set_csc(struct mxc_isi_dev *mxc_isi)
 	} else if (is_rgb(src_fmt->fourcc) && is_yuv(dst_fmt->fourcc)) {
 		/* RGB2YUV */
 		csc = RGB2YUV;
-		val |= (CHNL_IMG_CTRL_CSC_MODE_RGB2YUV << CHNL_IMG_CTRL_CSC_MODE_OFFSET);
+		val |= (CHNL_IMG_CTRL_CSC_MODE_RGB2YCBCR << CHNL_IMG_CTRL_CSC_MODE_OFFSET);
 	} else {
 		/* Bypass CSC */
 		printk("bypass csc\n");
-- 
1.7.9.5

